### PR TITLE
MAINT: Remove remaining uses of Python 3.6.

### PR DESCRIPTION
**Pull request by charris**
We missed a few first time around.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -26,8 +26,8 @@
 import warnings
 
 
-if sys.version_info[:2] < (3, 6):
-    raise RuntimeError("Python version >= 3.6 required.")
+if sys.version_info[:2] < (3, 7):
+    raise RuntimeError("Python version >= 3.7 required.")
```
Comment (mattip) [inline setup.py:30]: Is there a reason to actively reject 3.6 or is it enough to declare we no longer support it without forcing an error?

**Comment by mattip** [comment]
Since this will become an annual event, it may be helpful to document somewhere what needs to be done to add support for a new python version or remove support for an old one. Maybe somewhere around [here](https://numpy.org/devdocs/dev/releasing.html#supported-platforms-and-versions) ? It could be a follow-on PR if you want.

File: setup.py (hunk 1 above)
Comment (charris) [reply setup.py:30]: Not sure. Last I knew 1.20 still works with 3.6, we just don't want to officially release wheels for it and no longer test it. The minimal change for 1.20 would be to just remove 3.6 from `tox.ini` and leave everything else alone. OTOH, I believe there was a 3.6  incompatible change recently merged into master. So this is probably OK for master.

**Review by mattip — APPROVED** [review]

**Comment by charris** [comment]
Thinking about it, we don't have a good definition of what "drop support" means. Pip may get cheeky going forward as the metadata is stored in the wheels, so can be checked against the wheel name, but pip 20.3 is happy to build a wheel and install it with Python3.6, same with `setup.py` install. Removing 3.6 from the metadata doesn't seem to have any effect.

**Pull request merged**